Code/GraphMol/ChemReactions/Reaction.cpp

```diff
@@ -646,4 +646,18 @@ void ChemicalReaction::removeUnmappedProductTemplates(
   res_productTemplates.clear();
 }
 
+void ChemicalReaction::removeAgentTemplates(
+    MOL_SPTR_VECT *targetVector)
+{
+  if(targetVector){
```

Comment:
Should this be cleared first or is it ok to possibly add to whatever is already in the vector?